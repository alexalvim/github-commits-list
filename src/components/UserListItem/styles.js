import styled from 'styled-components';

import Spaces from '../../spaces';
import Typo from '../../typo';
import Colors from '../../colors';

const ContentWrapper = styled.div`
  align-items: center;
  background-color: ${Colors.lightestColor};
  color: ${Colors.darkestColor};
  cursor: pointer;
  display: flex;
  padding: ${Spaces.base};

  &:hover {
    background-color: ${Colors.mainColor};
    color: ${Colors.lightestColor};
  }
`;

const Title = styled.h3`
  font-size: ${Typo.medium};
`;

const Image = styled.img`
  border-radius: 50%;
  height: 40px;
  margin-right: ${Spaces.double};
  width: 40px;
`;

export {
  ContentWrapper,
  Title,
  Image
}