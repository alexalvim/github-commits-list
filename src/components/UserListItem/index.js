import React from 'react';

import { Title, Image, ContentWrapper } from './styles';

export default class UserListItem extends React.Component {
  render() {
    const { title, imgUrl } = this.props;
    return (
      <ContentWrapper>
        <Image src={imgUrl}/>
        <Title>{title}</Title>
      </ContentWrapper>)
  }
}