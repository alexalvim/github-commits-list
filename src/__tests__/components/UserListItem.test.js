import React from 'react';
import { shallow } from 'enzyme';

import UserListItem from '../../components/UserListItem';

describe('Testing UserListItem', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<UserListItem title='test' imgUrl='img' />);

    expect(wrapper).toMatchSnapshot();
  });
});