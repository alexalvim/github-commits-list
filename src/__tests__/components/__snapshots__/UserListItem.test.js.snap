// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UserListItem should render correctly 1`] = `ShallowWrapper {}`;
